libs/utils/WordPressUtils/src/main/java/org/wordpress/android/util/ImageUtils.java

```diff
@@ -545,47 +548,52 @@ public static String createResizedImageWithMaxWidth(Context context, String path
             return path;
         }
 
+        Uri srcImageUri = Uri.parse(path);
+        if (srcImageUri == null) {
+            return path;
+        }
+
         String fileName = MediaUtils.getMediaFileName(file, mimeType);
         String fileExtension = MimeTypeMap.getFileExtensionFromUrl(fileName).toLowerCase();
 
-        int[] dimensions = getImageSize(Uri.fromFile(file), context);
-        int orientation = getImageOrientation(context, path);
-
-        if (dimensions[0] <= maxWidth || maxWidth <= 0) {
-            // Image width is within limits; don't resize
+        int selectedWidth = getImageSize(srcImageUri, context)[0];
+        if (selectedWidth == 0) {
+            // Can't read the src dimensions.
             return path;
         }
-
-        Uri imageUri = Uri.parse(path);
-        if (context == null || imageUri == null) {
-            return path;
+        final int maxImageWidth = 3000;
```

Comment:
If there's any reason why we chose the number 3000, can we make it explicit in a comment please?